### ISSUE TITLE
Node Synchronization

### ISSUE THREAD

**Issue by Th-Pap** [taikoxyz/simple-taiko-node#172]
### Describe the bug

It has been observed that some nodes do not synchronize and the chain head on Grafana is shown as flatlined with no progress. Sometimes a restart of the node achieves some progress but not catching up with the chain. 
![kernel 65 a](https://github.com/taikoxyz/taiko-mono/assets/121475653/63deb00d-4d11-490e-981d-95960da608cf)
![kernel 65 b](https://github.com/taikoxyz/taiko-mono/assets/121475653/93b1b27e-2b46-4962-8151-3cadfd2d1680)
![node kernel 5 15](https://github.com/taikoxyz/taiko-mono/assets/121475653/52fc88d6-075d-48cb-84b7-3e06af2a7461)

I have been trying to identify the reason working in parallel with three nodes and using the same RPCs for all three. You can see the examples attached. The two flatlined nodes are running on two different servers both with Ubuntu Ubuntu 22.04.3 LTS (GNU/Linux 6.5.0-14-generic x86_64). The correct chain head is running on a server with Ubuntu 22.04.3 LTS (GNU/Linux 5.15.0-25-generic x86_64).

### Steps to reproduce

After identifying the possible issue and in an effort to double check the observation, I have downgraded one of the 6.5 kernel servers to kernel 5.15 and did a fresh install of docker, git and simple-taiko-node. The node started syncing and is achieving sync to the current block-required a couple of restarts though. Based on this analysis there is a possibility that nodes are not synchronizing due to an issue with kernel versions. This is an assumption based on observations only and may or may not be correct. However, I hope that this will assist the Taiko team to take the project a step further. Thank you

### Additional context

Additional context here.

**Comment by davidtaikocha** [comment]
Could u plz also show a screenshot of the CPU / memory usage of the node in grafana?

**Comment by Th-Pap** [comment]
![kernel 515 syncing](https://github.com/taikoxyz/simple-taiko-node/assets/121475653/3e977e5d-e734-43ae-a9bb-7848ebf92ac9)
Sure this node with kernel 5.15 syncing. The machine is a VPS 24c, 120GB ram.
Let me know if you want more info regarding the other node with kernel 6.5 NOT syncing.

**Comment by v54dt** [comment]
I have done several tests on different OS and different kernel version. All stuck during the synchronization and all uses a remote private holesky node. 

| VM      | Virtualization | OS | Kernel | Status |
| :---:        |    :----:   |:---: | :---: | :-----:|
| 1      | Kvm       | Debian 12   | 6.1.0-13-amd64|  A |
|    |         | Ubuntu 22.04.3| 5.15.0-46-generic| B |
|  || Ubuntu 22.04.3 | Linux 6.5.1-060500-generic| B |
|2| OpenVZ | Debian 12 | 6.1.0|A|
|||Ubuntu 22.04.3|5.2.0|A|
|||Ubuntu 22.04.6|5.4.0|A|

Status A : Stuck happened before node sync to latest verified block
Status B : Stuck happened after it has sync to latest verified block and switch to insert pending block one by one


Based on the screenshot below, I think the problem may be the one by one insert function handling the payload. [#164](https://github.com/taikoxyz/taiko-geth/pull/164)
![截圖 2024-01-30 凌晨12 34 23](https://github.com/taikoxyz/simple-taiko-node/assets/23051870/ead1a031-2d17-4f74-b997-cd63844b1aa5)


![截圖 2024-01-30 凌晨12 31 06](https://github.com/taikoxyz/simple-taiko-node/assets/23051870/09d7ad6b-9cdf-4504-a186-1e16b305ef76)
Here is my speculation about one by one function. The node just sync to 104,175, which is the latest verified block at the moment. And the sync mode has changed to insert one by one block, then the node start to stuck syncing.
![截圖 2024-01-30 下午3 25 08](https://github.com/taikoxyz/simple-taiko-node/assets/23051870/ce50c819-4c85-4250-a682-90c0a087a737)
Ram and disk usage drop due to manual node restart.

**Comment by Th-Pap** [comment]
Okay that's what I call thorough testing. I think that your speculation has some ground for additional testing. Let me know how I can help from my end

**Comment by davidtaikocha** [comment]
> Okay that's what I call thorough testing. I think that your speculation has some ground for additional testing. Let me know how I can help from my end

Just pushed some new images, could you give `docker compose down -v && docker compose up` another try?

**Comment by Th-Pap** [comment]
Done. I have executed docker compose down -v && docker compose up on three nodes simultaneously. I see that all three have the same logs. Let me revert back to you with the results in sometime. Let me know if you want additional details for each machine

**Comment by v54dt** [comment]
@davidtaikocha 
[https://github.com/taikoxyz/taiko-geth/commit/4d8479479d2fe82ed7f77e710ca9bcd70a07fd3a](https://github.com/taikoxyz/taiko-geth/commit/4d8479479d2fe82ed7f77e710ca9bcd70a07fd3a) The latest katla docker image fix the problem. Both works on kernel 5.x.x and 6.x.x and no matter is a fresh reinstall or just update the image.

**Comment by Th-Pap** [comment]
![node3](https://github.com/taikoxyz/simple-taiko-node/assets/121475653/310b32e7-8b57-4f4d-b53d-9c1ee1dad4e7)
![node2](https://github.com/taikoxyz/simple-taiko-node/assets/121475653/16d7810d-129c-437d-b56c-0872dd486338)
![node1](https://github.com/taikoxyz/simple-taiko-node/assets/121475653/e52eee42-44c1-4dec-9e2c-036dd585fe26)
![nodeinfo](https://github.com/taikoxyz/simple-taiko-node/assets/121475653/513586ab-2970-4842-a932-0283af95969b)
Okay, I am uploading info of the three nodes and snapshots. I had to restart node3 as it was stuck in a flatline.
I am happy to do an installation from scratch to all three of them (will take sometime for the holesky to sync) if you wish. However, if you want me to do that node3 is the sgx machine that I have generated the public key and registered on the form - not sure if this might have an impact.

**Comment by davidtaikocha** [comment]
> The two ![node1](https://private-user-images.githubusercontent.com/121475653/300887741-0662c9a8-3d63-4257-a187-e8ee86552cac.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2MzI1NTgsIm5iZiI6MTcwNjYzMjI1OCwicGF0aCI6Ii8xMjE0NzU2NTMvMzAwODg3NzQxLTA2NjJjOWE4LTNkNjMtNDI1Ny1hMTg3LWU4ZWU4NjU1MmNhYy5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEzMFQxNjMwNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNGRjMDIwYTMzNjljNjA0NDA5OTBhYTA1ZjU0MzE0NDE4OTRlMDE5MTRjNTdmOTA1MTczM2NlNGVmMDhkMGM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.9-XPHZFi7DFrN7ZltuP8e93VhfWYsVC2s57yL9Dwuyo) ![node2](https://private-user-images.githubusercontent.com/121475653/300887758-0a0e7a12-8ab8-4fc7-b967-4de5c43c76e1.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2MzI1NTgsIm5iZiI6MTcwNjYzMjI1OCwicGF0aCI6Ii8xMjE0NzU2NTMvMzAwODg3NzU4LTBhMGU3YTEyLThhYjgtNGZjNy1iOTY3LTRkZTVjNDNjNzZlMS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEzMFQxNjMwNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05M2M1ODk2ZWJkNzc4ODgzOTJjNzhkOTc3NGMzOTc5NDBmYzQ3OTRjYmNmOTRhMjA4NzMxNjYyOWUzOTBkZTg5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.EmRHF5sZa3BT-lLAxM7GdOTIB1ECzKMK-f7lD7UvWys) VPS nodes synced and looks good. I will delete simple-taiko-node on the dedicated and reinstall. I will let you know

Thanks for the report!

**Comment by davidtaikocha** [comment]
Closed, feel free to open it again when there is a new issue :)
